install-cloud.sh

```diff
@@ -11,27 +11,52 @@
 
 set -e
 
-if [ -f /etc/centos-release ]
-then 
-  yum update -y
-  yum install -y git bind-utils nmap-ncat
-  if [ ! -f /usr/bin/docker ]
+command_exists() {
+  command -v "$@" > /dev/null 2>&1
+}
+
+show_deps() {
+  echo "Before running this script, please install git, nc, and Docker."
+  echo "Installation instructions for Docker can be found here:"
+  echo "  https://docs.docker.com/mac/started/"
+  exit 1
+}
+
+do_install() {
+  # Since we cannot currently use DigitalOcean's Docker slug, install Docker
+  # if necessary. We also run a few other dependency checks to help those
+  # running on unsupported systems. Note that git is *not* present by default
+  # on DigitalOcean's Ubuntu 14.04 image: Docker's installer installs it. This
+  # is why we check for Docker first, even though it may seem cleaner to test
+  # for lighter-weight dependencies first.
+  if ! command_exists docker
   then
+    if [ "$USER" != "root" ]
+    then
+      show_deps
+    fi
+    echo "Docker not found, running Docker installer."
     curl -fsSL https://get.docker.com/ | sh
-    service docker start
-  elif [ -f /usr/bin/docker -a $(( $(docker --version | cut -d . -f 2) < 10 )) ]
+  fi
+  # We depend on --build-arg, introduced in Docker 1.9.
+  # Some distros are really slow to update.
+  if [ `docker version --format '{{.Server.Version}}' | cut -d . -f 2` -lt 9 ]
   then
-    echo "Installed version of docker is too old.  Please upgrade it yourself, or remove"
-    echo "it (yum erase docker) and run this script again.  If you don't have other"
-    echo "docker containers/images, removing it is fine, this script can do the rest. If"
-    echo "you do have other docker containers/images, have a look here:"
-    echo " https://github.com/docker/docker/wiki/Engine-v1.10.0-content-addressability-migration"
-    echo "first, and then upgrade docker yourself before running this script again."
+    echo "Before running this script, please upgrade Docker to version 1.9 or"
+    echo "greater. If you have containers or images you wish to use afterwards,"
+    echo "read this page first:"
+    echo "  https://github.com/docker/docker/wiki/Engine-v1.10.0-content-addressability-migration"
     exit 1
   fi
-fi
 
-do_install() {
+  for dep in git nc
```

Comment:
Tiny nit: put the list (git and nc) in a variable that you can print in show_deps.